pkg/ccl/backupccl/create_scheduled_backup.go

```diff
@@ -0,0 +1,466 @@
+	if eval.scheduleName != nil {
+		name, err := eval.scheduleName()
+		if err != nil {
+			return err
+		}
+		sj.SetScheduleName(name)
+	} else {
+		fmtCtx := tree.NewFmtCtx(tree.FmtSimple)
+		fmt.Fprintf(fmtCtx, "%s BACKUP OF ", args.BackupType)
+		if eval.Targets == nil {
+			fmtCtx.WriteString("CLUSTER")
+		} else {
+			eval.Targets.Format(fmtCtx)
+		}
+		sj.SetScheduleName(fmtCtx.String())
+	}
+
+	if eval.recurrence != nil {
+		recurrence, err := eval.recurrence()
+		if err != nil {
+			return err
+		}
+		err = sj.SetSchedule(recurrence)
+		if err != nil {
+			return err
+		}
+	}
+
+	if err := setScheduleOptions(eval, &p.ExtendedEvalContext().EvalContext, sj); err != nil {
+		return err
+	}
+
+	// Prepare backup statement.
+	backupNode := &tree.Backup{
+		Options: tree.BackupOptions{
+			CaptureRevisionHistory: eval.BackupOptions.CaptureRevisionHistory,
+			Detached:               true,
+		},
+	}
+
+	// Evaluate encryption passphrase if set.
+	if eval.encryptionPassphrase != nil {
+		pw, err := eval.encryptionPassphrase()
+		if err != nil {
+			return errors.Wrapf(err, "failed to evaluate backup encryption_passphrase")
+		}
+		backupNode.Options.EncryptionPassphrase = tree.NewDString(pw)
+	}
+
+	// Evaluate required backup destinations.
+	destinations, err := eval.destination()
+	if err != nil {
+		return errors.Wrapf(err, "failed to evaluate backup destination paths")
+	}
+	for _, dest := range destinations {
+		backupNode.To = append(backupNode.To, tree.NewDString(dest))
+	}
+
+	// Set backup targets and descriptor coverage.
+	if eval.Targets == nil {
+		backupNode.DescriptorCoverage = tree.AllDescriptors
+	} else {
+		backupNode.DescriptorCoverage = tree.RequestedDescriptors
+		backupNode.Targets = *eval.Targets
+	}
+
+	// TODO(yevgeniy): Validate backup schedule:
+	//  * Verify targets exist.  Provide a way for user to override this via option.
+	//  * Verify destination paths sane (i.e. valid schema://, etc)
+
+	// We do not set backupNode.AsOf: this is done when the scheduler kicks off the backup.
+	// Serialize backup statement and set schedule executor and its args.
+	args.BackupStatement = tree.AsString(backupNode)
+	any, err := pbtypes.MarshalAny(args)
+	if err != nil {
+		return err
+	}
+	sj.SetExecutionDetails(scheduledBackupExecutorName, jobspb.ExecutionArguments{Args: any})
+
+	// Create the schedule.
+	if err := p.ExecCfg().DB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
+		return sj.Create(ctx, p.ExecCfg().InternalExecutor, txn)
+	}); err != nil {
+		return err
+	}
+
+	var nextRun tree.Datum
+	if sj.IsPaused() {
+		nextRun = tree.DNull
+	} else {
+		next, err := tree.MakeDTimestampTZ(sj.NextRun(), time.Microsecond)
+		if err != nil {
+			return err
+		}
+		nextRun = next
+	}
+
+	resultsCh <- tree.Datums{
+		tree.NewDInt(tree.DInt(sj.ScheduleID())),
+		nextRun,
+		tree.NewDString(tree.AsString(eval.ScheduledBackup)),
+	}
+	return nil
+}
+
+const scheduleBackupOp = "SCHEDULED_BACKUP"
```

Comment:
Shouldn't this be `CREATE SCHEDULE FOR BACKUP`? Otherwise the errors below will read "argument of SCHEDULE_BACKUP must be .." which isn't a thing